website/docs/plugin/sdkv2/resources/data-consistency-errors.mdx

```diff
@@ -0,0 +1,112 @@
+
+</Note>
+
+Enable the `Resource` type `EnableApplyLegacyTypeSystemErrors` and `EnablePlanLegacyTypeSystemErrors` boolean fields and run all available [acceptance testing](/terraform/plugin/testing/acceptance-tests) for the resource. These settings will cause Terraform to raise data consistency errors instead of demoting those to warning logs for the resource.
+
+In this example, a resource has been enabled to raise data consistency errors instead of warning logs:
+
+```go
+schema.Resource{
+    // ... other fields as necessary ...
+    EnableApplyLegacyTypeSystemErrors: true,
+    EnablePlanLegacyTypeSystemErrors:  true,
+}
+```
+
+If there are unavoidable data consistency errors after reviewing the [Resolving Data Consistency Errors](#resolving-data-consistency-errors) section, disable only the setting(s) causing errors. This ensures that future changes to the resource are not introducing new data consistency issues.
+
+### Acceptance Testing All Values
+
+If the resource has unavoidable data consistency errors after reviewing the [Resolving Data Consistency Errors](#resolving-data-consistency-errors) section, a last resort option for checking one type of data consistency error is implementing [acceptance testing](/terraform/plugin/testing/acceptance-tests) that validates every configured attribute has the same exact value in state after being applied. The typical recommendation for state value checking in acceptance testing is only for `Computed: true` attributes, however since this SDK can bypass the Terraform data consistency rules, it is possible for state values to not match configuration values.
+
+In this example, an acceptance test performs the typically unnecessary state value check of a configured value:
+
+```go
+func TestThingResource(t *testing.T) {
+    resource.Test(t, resource.TestCase{
+        // ... other fields as necessary ...
+        Steps: []resource.TestStep{
+            {
+                // ... other fields as necessary ...
+                Config: `
+                resource "examplecloud_thing" "test" {
+                    configurable_attribute = "test-value"
+                }
+                `
+                Checks: resource.ComposeAggregateTestCheckFunc(
+                    resource.TestCheckResourceAttr("examplecloud_thing.test", "configurable_attribute", "test-value"),
+                ),
+            },
+            // ... potentially other steps as necessary ...
+        },
+    })
+}
+```
+
+The state value check will raise an error if there is an unexpected value difference, which is the same as if Terraform raised the error. Checking state values of configured values can be removed once the resource is [migrated to terraform-plugin-framework](/terraform/plugin/framework/migrating), as Terraform itself will raise any potential error.
+
+## Resolving Data Consistency Errors
```

Comment:
Not sure we need to open the can of worms in this PR, but should we eventually add the other common error `planned value does not match config value or prior value` to this section?

Reply:
Yeah, that would eventually be my goal from the PR description. I just don't have all those various errors handy right now and we'll need to figure out how we best want to discuss each one.

Reply:
Sorry! I missed that part of your PR description 😵‍💫